WordPress/gutenberg
# Load block support styles in the `<head>` for block themes.

Part of https://github.com/WordPress/gutenberg/issues/38434
Related https://github.com/WordPress/gutenberg/pull/38745
Follow-up to https://github.com/WordPress/gutenberg/pull/32083

The dynamic block styles for layout and elements should be loaded in the `<head>` for block themes. While that should also be the case for classic themes, the current mechanism we use (`render_block`) does not allow us to do that, hence, this PR doesn't change anything for them and will be loaded the `<body>`.

## How to test

Block theme in WordPress 5.9:

- Use TwentyTwentyTwo and load the front end.
- Verify the embedded styles for elements (`.wp-elements-UUID ...`) and layout (`.wp-container-UUUID ...`) are loaded in the `<head>`.

Classic theme in WordPress 5.9:

- Use TwentyTwenty and load the front end.
- Verify the embedded styles for elements (`.wp-elements-UUID ...`) and layout (`.wp-container-UUUID ...`) are loaded in the `<body>`.
